cql3/statements/describe_statement.cc

```diff
@@ -160,14 +161,31 @@ description type(replica::database& db, const lw_shared_ptr<keyspace_metadata>&
     return description(db, *udt, true);
 }
 
-future<std::vector<description>> types(replica::database& db, const lw_shared_ptr<keyspace_metadata>& ks, bool with_stmt = false) {
-    auto user_types_map = ks->user_types().get_all_types();
-    auto user_types = boost::copy_range<std::vector<shared_ptr<const keyspace_element>>>(user_types_map | boost::adaptors::transformed([] (const auto& e) {
-        // return static_pointer_cast<const keyspace_element>(e.second);
-        return e.second;
-    }));
+// Because UDTs can depend on each other, we need to sorte the topologically
+future<std::vector<shared_ptr<const keyspace_element>>> get_sorted_types(const lw_shared_ptr<keyspace_metadata>& ks) {
+    struct udts_comparator {
+        inline bool operator()(const user_type& a, const user_type& b) const {
+            return a->get_name_as_string() < b->get_name_as_string();
+        }
+    };
+
+    std::vector<user_type> all_udts;
+    std::multimap<user_type, user_type, udts_comparator> adjacency;
 
-    co_return co_await generate_descriptions(db, user_types, (with_stmt) ? std::optional(true) : std::nullopt);
+    for (auto& [_, udt]: ks->user_types().get_all_types()) {
+        all_udts.push_back(udt);
+        for (auto& ref_udt: udt->get_all_referenced_user_types()) {
+            adjacency.insert({ref_udt, udt});
+        }
```

Comment:
Nice, this way we don't have to reverse the vector at the end like we used to have in the previous version of the implementation.